### PR TITLE
Make ChangePassword a real Controller

### PR DESCRIPTION
**Pull request by rullzer**
Less depreacted calls :tada: 

This will get us a step closer to properly changepassword stuff.

CC: @LukasReschke @nickvergessen @MorrisJobke @icewind1991

### CODE REVIEW

File: settings/Controller/ChangePasswordController.php [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	/** @var IGroupManager */
+	private $groupManager;
+
+	/** @var IUserSession */
+	private $userSession;
+
+	/** @var IAppManager */
+	private $appManager;
+
+	public function __construct($appName,
```
Comment (LukasReschke) [inline settings/Controller/ChangePasswordController.php]: PHPDoc? ;)

**Comment by LukasReschke** [comment]
Mind adding some tests for the stuff that's testable? (e.g. changePersonalPassword)

File: settings/Controller/ChangePasswordController.php (hunk 1 above)
Comment (rullzer) [inline settings/Controller/ChangePasswordController.php]: Done

**Comment by rullzer** [comment]
Added some tests

**Comment by LukasReschke** [comment]
LGTM

File: settings/Controller/ChangePasswordController.php [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+		}
+
+		return new JSONResponse([
+			'status' => 'success',
+			'data' => [
+				'username' => $username,
+			],
+		]);
+	}
+}
```
Comment (nickvergessen) [inline settings/Controller/ChangePasswordController.php]: Fix your IDE finally

**Comment by nickvergessen** [comment]
Changing still works as user, admin and subadmin

:+1:

**Comment by nickvergessen** [comment]
> This branch has conflicts that must be resolved

Please rebase while fixing the typo from above

File: settings/Controller/ChangePasswordController.php (hunk 2 above)
Comment (rullzer) [inline settings/Controller/ChangePasswordController.php]: fixed

**Comment by rullzer** [comment]
And rebased

**Comment by MorrisJobke** [comment]
There was a regression: #1634 fixed it ;)

**Pull request merged**